core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleManager.scala

```diff
@@ -187,6 +187,7 @@ private[spark] class SortShuffleManager(conf: SparkConf) extends ShuffleManager
         shuffleBlockResolver.removeDataByMap(shuffleId, mapTaskId)
       }
     }
+    shuffleBlockResolver.removeShuffleToSkip(shuffleId)
```

Comment:
this is a weird place to do cleanup. Shall we cover all shuffle manager implementations? Shall we do it in the caller of this `unregisterShuffle` function?